examples/2d/sprite_slice.rs

```diff
@@ -0,0 +1,171 @@
+            sides_scale_mode: SliceScaleMode::Tile { stretch_value: 0.3 },
+            ..default()
+        }),
+        ..default()
+    });
+
+    // Scaled sliced sprite horizontally with max scale
+    commands.spawn(SpriteBundle {
+        transform: Transform::from_translation(base_pos + Vec3::X * 1050.0),
+        texture: texture_handle,
+        sprite: Sprite {
+            custom_size: Some(Vec2::new(300.0, 200.0)),
+            ..default()
+        },
+        scale_mode: ImageScaleMode::Sliced(TextureSlicer {
+            border: BorderRect::square(slice_border),
+            center_scale_mode: SliceScaleMode::Tile { stretch_value: 0.1 },
+            sides_scale_mode: SliceScaleMode::Tile { stretch_value: 0.2 },
+            max_corner_scale: 0.2,
+        }),
+        ..default()
+    });
+}
+
+fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
+    commands.spawn(Camera2dBundle::default());
+    // Load textures
+    let handle_1 = asset_server.load("textures/slice_square.png");
+    let handle_2 = asset_server.load("textures/slice_square_2.png");
+
+    spawn_sprites(
+        &mut commands,
+        handle_1,
+        Vec3::new(-600.0, 200.0, 0.0),
+        200.0,
+    );
+    spawn_sprites(
+        &mut commands,
+        handle_2,
+        Vec3::new(-600.0, -200.0, 0.0),
+        80.0,
+    );
+
+    let font = asset_server.load("fonts/FiraSans-Bold.ttf");
+    let style = TextStyle {
+        font: font.clone(),
+        font_size: 30.0,
+        color: Color::WHITE,
+    };
+    let alignment = JustifyText::Center;
+    // Spawn text
+    commands.spawn(Text2dBundle {
+        text: Text::from_section("Original texture", style.clone()).with_justify(alignment),
+        transform: Transform::from_xyz(-550.0, 0.0, 0.0),
+        ..default()
+    });
+    commands.spawn(Text2dBundle {
+        text: Text::from_section("Stretched texture", style.clone()).with_alignment(alignment),
+        transform: Transform::from_xyz(-400.0, 0.0, 0.0),
+        ..default()
+    });
+    commands.spawn(Text2dBundle {
+        text: Text::from_section("Stretched and sliced", style.clone()).with_alignment(alignment),
+        transform: Transform::from_xyz(-250.0, 0.0, 0.0),
+        ..default()
+    });
+    commands.spawn(Text2dBundle {
+        text: Text::from_section("Sliced and Tiled", style.clone()).with_alignment(alignment),
+        transform: Transform::from_xyz(-100.0, 0.0, 0.0),
+        ..default()
+    });
+    commands.spawn(Text2dBundle {
+        text: Text::from_section("Sliced and Tiled", style.clone()).with_alignment(alignment),
+        transform: Transform::from_xyz(150.0, 0.0, 0.0),
+        ..default()
+    });
+    commands.spawn(Text2dBundle {
+        text: Text::from_section("Sliced and Tiled with corner constraint", style.clone())
+            .with_alignment(alignment),
+        transform: Transform::from_xyz(550.0, 0.0, 0.0),
+        ..default()
+    });
+}
```

Comment:
To fix the text alignment problems you could make the labels relatively positioned children of the sprites:

```suggestion
fn spawn_sprites(
    commands: &mut Commands,
    texture_handle: Handle<Image>,
    mut position: Vec3,
    slice_border: f32,
    style: TextStyle,
    gap: f32,
) {
    let cases = [
        // Reference sprite
        (
            "Original texture",
            style.clone(),
            Vec2::splat(100.0),
            ImageScaleMode::default(),
        ),
        // Scaled regular sprite
        (
            "Stretched texture",
            style.clone(),
            Vec2::new(100.0, 200.0),
            ImageScaleMode::default(),
        ),
        // Stretched Scaled sliced sprite
        (
            "Stretched and sliced",
            style.clone(),
            Vec2::new(100.0, 200.0),
            ImageScaleMode::Sliced(TextureSlicer {
                border: BorderRect::square(slice_border),
                center_scale_mode: SliceScaleMode::Stretch,
                ..default()
            }),
        ),
        // Scaled sliced sprite
        (
            "Sliced and Tiled",
            style.clone(),
            Vec2::new(100.0, 200.0),
            ImageScaleMode::Sliced(TextureSlicer {
                border: BorderRect::square(slice_border),
                center_scale_mode: SliceScaleMode::Tile { stretch_value: 0.5 },
                sides_scale_mode: SliceScaleMode::Tile { stretch_value: 0.2 },
                ..default()
            }),
        ),
        // Scaled sliced sprite horizontally
        (
            "Sliced and Tiled",
            style.clone(),
            Vec2::new(300.0, 200.0),
            ImageScaleMode::Sliced(TextureSlicer {
                border: BorderRect::square(slice_border),
                center_scale_mode: SliceScaleMode::Tile { stretch_value: 0.2 },
                sides_scale_mode: SliceScaleMode::Tile { stretch_value: 0.3 },
                ..default()
            }),
        ),
        // Scaled sliced sprite horizontally with max scale
        (
            "Sliced and Tiled with corner constraint",
            style,
            Vec2::new(300.0, 200.0),
            ImageScaleMode::Sliced(TextureSlicer {
                border: BorderRect::square(slice_border),
                center_scale_mode: SliceScaleMode::Tile { stretch_value: 0.1 },
                sides_scale_mode: SliceScaleMode::Tile { stretch_value: 0.2 },
                max_corner_scale: 0.2,
            }),
        ),
    ];

    for (label, text_style, size, scale_mode) in cases {
        position.x += 0.5 * size.x;
        commands
            .spawn(SpriteBundle {
                transform: Transform::from_translation(position),
                texture: texture_handle.clone(),
                sprite: Sprite {
                    custom_size: Some(size),
                    ..default()
                },
                scale_mode,
                ..default()
            })
            .with_children(|builder| {
                builder.spawn(Text2dBundle {
                    text: Text::from_section(label, text_style).with_justify(JustifyText::Center),
                    transform: Transform::from_xyz(0., -0.5 * size.y - 10., 0.0),
                    text_anchor: bevy::sprite::Anchor::TopCenter,
                    ..default()
                });
            });
        position.x += 0.5 * size.x + gap;
    }
}

fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
    commands.spawn(Camera2dBundle::default());
    let font = asset_server.load("fonts/FiraSans-Bold.ttf");
    let style = TextStyle {
        font: font.clone(),
        font_size: 16.0,
        color: Color::WHITE,
    };

    // Load textures
    let handle_1 = asset_server.load("textures/slice_square.png");
    let handle_2 = asset_server.load("textures/slice_square_2.png");

    spawn_sprites(
        &mut commands,
        handle_1,
        Vec3::new(-600.0, 200.0, 0.0),
        200.0,
        style.clone(),
        50.,
    );

    spawn_sprites(
        &mut commands,
        handle_2,
        Vec3::new(-600.0, -200.0, 0.0),
        80.0,
        style,
        50.,
    );
}

```